component/src/main/java/com/synopsys/integration/alert/component/settings/proxy/database/accessor/SettingsProxyConfigAccessor.java

```diff
@@ -0,0 +1,154 @@
+                                                   .collect(Collectors.toList());
+        return new AlertPagedModel<>(resultPage.getTotalPages(), resultPage.getNumber(), resultPage.getSize(), pageContent);
+    }
+
+    @Override
+    @Transactional(propagation = Propagation.REQUIRED)
+    public SettingsProxyModel createConfiguration(SettingsProxyModel configuration) {
+        OffsetDateTime currentTime = DateUtils.createCurrentDateTimestamp();
+        UUID configurationId = UUID.randomUUID();
+        configuration.setId(configurationId.toString());
+        SettingsProxyConfigurationEntity configurationToSave = toEntity(configurationId, configuration, currentTime, currentTime);
+        SettingsProxyConfigurationEntity savedProxyConfig = settingsProxyConfigurationRepository.save(configurationToSave);
+        List<NonProxyHostConfigurationEntity> nonProxyHosts = toNonProxyHostEntityList(configurationId, configuration);
+        nonProxyHostsConfigurationRepository.saveAll(nonProxyHosts);
+        savedProxyConfig = settingsProxyConfigurationRepository.getOne(savedProxyConfig.getConfigurationId());
+
+        return createConfigModel(savedProxyConfig);
+    }
+
+    @Override
+    @Transactional(propagation = Propagation.REQUIRED)
+    public SettingsProxyModel updateConfiguration(UUID configurationId, SettingsProxyModel configuration) throws AlertConfigurationException {
+        SettingsProxyConfigurationEntity configurationEntity = settingsProxyConfigurationRepository.findById(configurationId)
+                                                                   .orElseThrow(() -> new AlertConfigurationException(String.format("Config with id '%d' did not exist", configurationId)));
+        OffsetDateTime currentTime = DateUtils.createCurrentDateTimestamp();
+        SettingsProxyConfigurationEntity configurationToSave = toEntity(configurationId, configuration, configurationEntity.getCreatedAt(), currentTime);
+        SettingsProxyConfigurationEntity savedProxyConfig = settingsProxyConfigurationRepository.save(configurationToSave);
+        List<NonProxyHostConfigurationEntity> nonProxyHosts = toNonProxyHostEntityList(configurationId, configuration);
+        nonProxyHostsConfigurationRepository.saveAll(nonProxyHosts);
+        savedProxyConfig = settingsProxyConfigurationRepository.getOne(savedProxyConfig.getConfigurationId());
+
+        return createConfigModel(savedProxyConfig);
+    }
+
+    @Override
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void deleteConfiguration(UUID configurationId) {
+        if (null != configurationId) {
+            settingsProxyConfigurationRepository.deleteById(configurationId);
+        }
+    }
+
+    private SettingsProxyModel createConfigModel(SettingsProxyConfigurationEntity proxyConfiguration) {
+        SettingsProxyModel newModel = new SettingsProxyModel();
+        String createdAtFormatted = "";
+        String lastUpdatedFormatted = "";
```

Comment:
These may not be necessary. You should be able to move the set call into the if statement and a null value should set a default once inserted into the DB